File: java/121.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = prices.length;
        if(l < 2){
            return 0;
        }
        int buy = 0;
        int ans = 0;
        for(int i = 0 ; i < l;i++){
            ans = Math.max(prices[i] - prices[buy],ans);
            if(prices[buy] > prices[i]){
                buy = i;
            }
        }
        return ans;
        
    }
}

File: go/121.go
func maxProfit(prices []int) int {
	min := int(^uint(0) >> 1)
	prof := 0
	for i := 0; i < len(prices); i++ {
		if prices[i] < min {
			min = prices[i]
		} else if prices[i]-min > prof {
			prof = prices[i] - min
		}
	}
	return prof

}